acvm-repo/brillig/src/opcodes.rs

```diff
@@ -301,20 +316,26 @@ pub enum BrilligOpcode<F> {
         source_b: MemoryAddress,
         condition: MemoryAddress,
     },
+    /// Reads the `source_pointer` to obtain a memory address, then retrieves the data
+    /// stored at that address and writes it to the `destination` address.
     Load {
         destination: MemoryAddress,
         source_pointer: MemoryAddress,
     },
+    /// Reads the `destination_pointer` to obtain a memory address, then stores the value
+    /// from the `source` address at that location.
     Store {
         destination_pointer: MemoryAddress,
         source: MemoryAddress,
     },
+    /// Native functions in the VM.
+    /// These are equivalent to the black box functions in ACIR.
     BlackBox(BlackBoxOp),
-    /// Used to denote execution failure, returning data after the offset
+    /// Used to denote execution failure, halting the VM and returning data specified by a memory-sized vector.
     Trap {
         revert_data: HeapVector,
     },
-    /// Stop execution, returning data after the offset
+    /// Halts execution and returns data specified by a memory-sized vector.
```

Comment:
What does it mean to be `memory-sized`? Will this opcode fail if the `HeapVector` is of a different size?

Reply:
`memory-sized` just means the [size of the vector is specified dynamically in memory](https://github.com/noir-lang/noir/blob/c757112adc707a65bddc54b71dbe4e11c477e2bf/acvm-repo/brillig/src/opcodes.rs#L103) vs. a fixed-sized array. I could change this to a "dynamic vector" if that is clearer? Not sure what you mean by "of a different size"?

Reply:
Yes, please change to "dynamic" because it sounded like a constant

Reply:
Switched to `dynamically-sized`